internal/engine/local_container_build_and_deployer.go

```diff
@@ -88,31 +88,22 @@ func (cbd *LocalContainerBuildAndDeployer) BuildAndDeploy(ctx context.Context, m
 	}, nil
 }
 
-func (cbd *LocalContainerBuildAndDeployer) PostProcessBuilds(ctx context.Context, states BuildStatesByName) {
-	span, ctx := opentracing.StartSpanFromContext(ctx, "LocalContainerBuildAndDeployer-PostProcessBuilds")
+func (cbd *LocalContainerBuildAndDeployer) PostProcessBuild(ctx context.Context, manifest model.Manifest, result BuildResult) {
+	span, ctx := opentracing.StartSpanFromContext(ctx, "LocalContainerBuildAndDeployer-PostProcessBuild")
+	span.SetTag("manifest", manifest.Name.String())
 	defer span.Finish()
 
-	// HACK(maia): give the pod(s) we just deployed a bit to come up.
-	// TODO(maia): replace this with polling/smart waiting
-	logger.Get(ctx).Infof("Post-processing %d builds...", len(states))
-
-	for name, state := range states {
-		cbd.postProcessBuild(ctx, name, state)
-	}
-}
-
-func (cbd *LocalContainerBuildAndDeployer) postProcessBuild(ctx context.Context, name model.ManifestName, state BuildState) {
-	if !state.LastResult.HasImage() {
-		logger.Get(ctx).Infof("can't get container for %q: BuildResult has no image", name)
+	if !result.HasImage() {
+		logger.Get(ctx).Infof("can't get container for for '%s': BuildResult has no image", manifest.Name)
 		return
 	}
-	if _, ok := cbd.deployInfo[name]; !ok {
-		cID, err := cbd.getContainerForBuild(ctx, state.LastResult)
+	if _, ok := cbd.deployInfo[manifest.Name]; !ok {
```

Comment:
if you want this to be thread-safe you need a mutex whenever you access cbd.deployInfo?